app/javascript/components/venue/VenueEdit.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react';
+import Axios from 'axios';
+import Grid from '@material-ui/core/Grid';
+import { FormTextField } from '../../ui/TextFileds';
+import { FormButton } from '../../ui/Buttons';
+
+export default class VenueEdit extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      venue: { address: '', description: '', people_capacity: '' }
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange = (venue) => {
+    var prevState = {...this.state};
+    var updatedVenue = {...this.state.venue};
+    updatedVenue[venue.target.name] = venue.target.value;
+    this.setState({ ...prevState, venue: updatedVenue });
+  }
+
+  handleSubmit = venue => {
+    venue.preventDefault();
+    let headers = {};
+    if (sessionStorage.user) {
+      headers = JSON.parse(sessionStorage.user);
+    }
+    headers['X-CSRF-Token'] = document
+      .querySelector('meta[name=\'csrf-token\']')
+      .getAttribute('content');
+    Axios.patch(
+      `/api/v1/venues/${this.props.match.params.id}`,
+      { venue: this.state.venue },
+      { headers: headers }
+    )
+      .then(() => this.props.history.push('/venues'))
+      .catch(error => {
+        if (error.response.statusText === 'Locked')
+          this.setState({ errors: ['Only orgnizer can do this'] });
```

Comment:
typo

Reply:
I still think it's better to retrieve error message from backend rather then rely on status code and generate it on frontend

Reply:
done